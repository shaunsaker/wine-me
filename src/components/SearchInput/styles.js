import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    alignSelf: 'stretch',
    backgroundColor: styleConstants.colors.transBlack,
    paddingVertical: styleConstants.dimensions.spacing.vertical,
    paddingHorizontal: styleConstants.dimensions.spacing.horizontal,
    borderRadius: styleConstants.dimensions.borderRadius,
    flexDirection: 'row',
    alignItems: 'center',
  },
  icon: {
    fontSize: styleConstants.fonts.sizes.icon,
    color: styleConstants.colors.white,
    marginRight: styleConstants.dimensions.spacing.horizontal / 2,
  },
  input: {
    flex: 1,
  },
  text: {
    ...styleConstants.fonts.types.paragraph,
    color: styleConstants.colors.white,
  },
});

export default styles;
