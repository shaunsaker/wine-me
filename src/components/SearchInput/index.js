import React from 'react';
import PropTypes from 'prop-types';
import { TextInput, Text, ViewPropTypes } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styleConstants from '../../styleConstants';
import styles from './styles';

import Touchable from '../Touchable';

const propTypes = {
  handlePress: PropTypes.func, // if supplied will render Search text instead of an input
  value: PropTypes.string,
  handleChangeText: PropTypes.func,
  autoFocus: PropTypes.bool,
  style: ViewPropTypes.style,
};

const defaultProps = {};

const SearchInput = ({ handlePress, value, handleChangeText, autoFocus, style }) => {
  const textComponent = handlePress ? (
    <Text style={styles.text}>Search</Text>
  ) : (
    <TextInput
      placeholder="Search"
      placeholderTextColor={styleConstants.colors.white}
      value={value}
      onChangeText={handleChangeText}
      autoFocus={autoFocus}
      style={[styles.input, styles.text]}
    />
  );

  return (
    <Touchable onPress={handlePress} disabled={!handlePress} style={[styles.container, style]}>
      <Icon name="search" style={styles.icon} />

      {textComponent}
    </Touchable>
  );
};

SearchInput.propTypes = propTypes;
SearchInput.defaultProps = defaultProps;

export default SearchInput;
