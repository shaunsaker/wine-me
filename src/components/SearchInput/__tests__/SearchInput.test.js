import React from 'react';
import renderer from 'react-test-renderer';

import SearchInput from '..';

describe('SearchInput', () => {
  const handlePress = jest.fn();

  describe('renders', () => {
    it('renders with the minimum required props', () => {
      const component = renderer.create(<SearchInput />);

      expect(component).toMatchSnapshot();
    });

    it('renders the handlePress state', () => {
      const component = renderer.create(<SearchInput handlePress={handlePress} />);

      expect(component).toMatchSnapshot();
    });
  });
});
