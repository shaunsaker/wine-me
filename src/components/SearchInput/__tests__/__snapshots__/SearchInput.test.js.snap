// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders renders the handlePress state 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "rgba(0, 0, 0, 0.70)",
      "borderRadius": 5,
      "flexDirection": "row",
      "height": 48,
      "opacity": 1,
      "paddingHorizontal": 15,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": "#FFFFFF",
          "fontSize": 24,
          "marginRight": 7.5,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 17,
      }
    }
  >
    Search
  </Text>
</View>
`;

exports[`SearchInput renders renders with the minimum required props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "rgba(0, 0, 0, 0.70)",
      "borderRadius": 5,
      "flexDirection": "row",
      "height": 48,
      "opacity": 1,
      "paddingHorizontal": 15,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": "#FFFFFF",
          "fontSize": 24,
          "marginRight": 7.5,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <TextInput
    allowFontScaling={true}
    placeholder="Search"
    placeholderTextColor="#FFFFFF"
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "color": "#FFFFFF",
          "fontSize": 17,
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
